spec/datadog/integration_spec.rb

```diff
@@ -23,22 +23,30 @@ def wait_for_tracer_sent
 
     context 'for threads' do
       let(:original_threads) { Thread.list }
-      let!(:original_threads_inspect) { inspect_threads(original_threads) } # Store result as stack trace will change
-      let(:threads) { Thread.list }
+      let(:start_tracer) do
+        Datadog::Tracing.trace('test.op') {}
+
+        new_threads = Thread.list - original_threads
+        tracer_threads.concat(new_threads.map(&:object_id))
+      end
+      let(:tracer_threads) { [] }
 
       def inspect_threads(threads)
-        threads.map.with_index { |t, idx| "#{idx}=#{t.backtrace}" }.join(';')
+        threads.map.with_index { |t, idx| "#{idx}=#{t.object_id}:#{t.backtrace}" }.join(';')
       end
 
+      subject(:shutdown) { Datadog.shutdown! }
+
       it 'closes tracer threads' do
         start_tracer
         wait_for_tracer_sent
 
         shutdown
 
-        expect(threads.size).to eq(original_threads.size),
-          "Expected #{original_threads.size} threads (#{original_threads_inspect}), "\
-                  "got #{threads.size} threads (#{inspect_threads(threads)})"
+        post_shutdown_threads = Thread.list.map(&:object_id)
+
+        expect(post_shutdown_threads & tracer_threads).to be_empty,
+          "Tracer threads not terminated: #{inspect_threads(Thread.list)}"
```

Comment:
I would suggest calling to `Thread.list` only once, because otherwise if a thread finishes between the call on line 46 and the call on line 49, we'll get a really confusing error message.